### PR TITLE
FailureStorage accepts FSA to avoid real file creation on in-memory fs

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

**Comment by jotomo** [comment]
@tinwelint This is just another quick idea from glancing at the changes: this seems to be in the same area as #1873 (schema index created on disk for in-memory graphdb). Thought it might be easy to fix while in the area. Might even be fixed by this, seeing that LuceneSchemaIndexProvider's ctor now takes a FileSystemAbstraction (if so, add a test to assert the issue is fixed)?

**Comment by MishaDemianenko** [comment]
@tinwelint there are test failures

**Comment by tinwelint** [comment]
Sorry, I've dropped the ball on this. Will try to update soon.

File: community/kernel/src/test/java/org/neo4j/kernel/api/index/util/FailureStorageTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private FolderLayout folderLayout;
+    private final long indexId = 1;
+
+    @Before
+    public void before()
+    {
+        File rootDirectory = new File( "dir" );
+        fs.get().mkdirs( rootDirectory );
+        folderLayout = new FolderLayout( rootDirectory );
+        fs.get().mkdirs( folderLayout.getFolder( indexId ) );
```
Comment (MishaDemianenko) [inline community/kernel/src/test/java/org/neo4j/kernel/api/index/util/FailureStorageTest.java]: We should not do it here right? `FailureStorage` should be able to create required dirs itself during `reserveForIndex` call

**Comment by MishaDemianenko** [comment]
@tinwelint only one small comment otherwise ok

**Comment by tinwelint** [comment]
Removed the additional `mkdirs` from `FailureStorageTest#before`, but kept the mkdirs for the root directory, because that's how it's going to look like anyway in a real environment.

**Comment by MishaDemianenko** [comment]
@tinwelint yep exactly, looks fine. Will merge as soon as CI build will be ready.

**Pull request merged**